cider/tests/utils/CiderBatchBuilder.h

```diff
@@ -28,6 +28,8 @@
 #include "substrait/type.pb.h"
 #include "util/Logger.h"
 
+static const std::shared_ptr<CiderAllocator> allocator =
```

Comment:
I am afraid it's not a good manner to define and initialize a static variable in a header file. Image what if you include this header in `CiderArrowBatchTest.cpp` which also define the same variable name.

Reply:
>  but in QueryDataGenerator.h there are some static function need to allocate memory, may be we can handle it only in way
you can define local variable just for function parameter

Reply:
> but in QueryDataGenerator.h there are some static function need to allocate memory, may be we can handle it only in way

you can define local variable just for function parameter

Reply:
done